nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PublishSlack.java

```diff
@@ -0,0 +1,513 @@
+        .required(true)
+        .defaultValue("UTF-8")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .dependsOn(PUBLISH_STRATEGY, PUBLISH_STRATEGY_CONTENT_AS_MESSAGE)
+        .build();
+
+    protected PropertyDescriptor MESSAGE_TEXT = new PropertyDescriptor.Builder()
+        .name("Message Text")
+        .description("The text of the message to send to Slack.")
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .required(true)
+        .addValidator(Validator.VALID)
+        .dependsOn(PUBLISH_STRATEGY, PUBLISH_STRATEGY_USE_PROPERTY)
+        .build();
+
+    protected PropertyDescriptor SEND_CONTENT_AS_ATTACHMENT = new PropertyDescriptor.Builder()
+        .name("Include FlowFile Content as Attachment")
+        .description("Specifies whether or not the contents of the FlowFile should be uploaded as an attachment to the Slack message.")
+        .allowableValues("true", "false")
+        .required(true)
+        .dependsOn(PUBLISH_STRATEGY, PUBLISH_STRATEGY_USE_PROPERTY)
+        .defaultValue("false")
+        .build();
+
+    protected PropertyDescriptor MAX_FILE_SIZE = new PropertyDescriptor.Builder()
+        .name("Max FlowFile Size")
+        .description("The maximum size of a FlowFile that can be sent to Slack. If any FlowFile exceeds this size, it will be routed to failure. " +
+                     "This plays an important role because the entire contents of the file must be loaded into NiFi's heap in order to send the data " +
+                     "to Slack.")
+        .required(true)
+        .dependsOn(SEND_CONTENT_AS_ATTACHMENT, "true")
+        .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+        .defaultValue("1 MB")
+        .build();
+
+    protected PropertyDescriptor THREAD_TS = new PropertyDescriptor.Builder()
+        .name("Thread Timestamp")
+        .description("The Timestamp identifier for the thread that this message is to be a part of. If not specified, the message will be a top-level message instead of " +
+                     "being in a thread.")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .required(false)
+        .build();
+
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("FlowFiles are routed to success after being successfully sent to Slack")
+        .build();
+
+    public static final Relationship REL_RATE_LIMITED = new Relationship.Builder()
+        .name("rate limited")
+        .description("FlowFiles are routed to 'rate limited' if the Rate Limit has been exceeded")
+        .build();
+
+    public static final Relationship REL_TOO_LARGE = new Relationship.Builder()
+        .name("too large")
```

Comment:
Instead of `too large`, what do you think about `size limit exceeded`, `size exceeded`, or perhaps `size limited` aligning with `rate limited`?